src/modules/navigator/mission.cpp

```diff
@@ -73,6 +76,12 @@ Mission::on_inactive()
 	 * is used for missions such as RTL. */
 	_navigator->set_cruising_speed();
 
+	// if we were executing an landing but have been inactive for 2 seconds, then make the landing invalid
+	// this prevents RTL to just continue at the current mission index
+	if (_navigator->getMissionLandingInProgress() && (hrt_absolute_time() - _time_mission_deactivated) > 2_s) {
```

Comment:
I'm not sure I follow, how would this fail if the mission landing progress was set to false immediately on first inactivation?

Reply:
@dagar This prevents the situation where e.g. you are on a mission landing and then RTL kicks in. You don't want to start from the beginning but continue the mission. However, if e.g. the user takes over in manual during a mission landing and then switches back, you probably want to start he mission landing from scratch, as you have no clue where exactly you are and if you are still aligned with the path.

Reply:
I should have been more specific, the part I don't follow is the magic 2 seconds.

Reply:
@dagar A long study over many years has calculated this number to be the one and only reasonable value. It's almost like pi....

Reply:
Why not do it immediately on the actual mission deactivation?

Reply:
This is for mission -> RTL (failsafe, etc) -> mission (planned landing)?

Reply:
Let's track this in a separate issue - we shouldn't be blocking an incremental move towards a much better behavior.

Reply:
Issue to track that: https://github.com/PX4/PX4-Autopilot/issues/16488

Reply:
@dagar I think I might need to explain this a bit better:
When you are executing a mission landing and RTL kicks in due to low battery, then the mode switches from mission to RTL and back to mission (assuming you are executing a mission landing, which is the preferred choice almost always).
Two the second delay (it actually needs way less) allows the mission to just continue the landing rather than restarting the landing pattern.

Reply:
@dagar Sorry, I noticed I told you this before. What is the issue here?

Reply:
> @dagar Sorry, I noticed I told you this before. What is the issue here?

Nothing necessarily, I just wanted to make sure there wasn't a potential hole for a hard to reproduce bug down the round. Eg dataman chokes reading the sdcard and there's a one off delay that's much longer than expected. 
￼